pkgs/servers/sql/postgresql/ext/omnigres.nix

```diff
@@ -0,0 +1,118 @@
+{
+  buildEnv,
+  bison,
+  boost,
+  brotli,
+  bzip2,
+  cmake,
+  doxygen,
+  fetchFromGitHub,
+  flex,
+  git,
+  lib,
+  libcap,
+  libtool,
+  libuv,
+  libxml2,
+  lld,
+  netcat,
+  openssl,
+  perl,
+  pkg-config,
+  postgresql,
+  postgresqlBuildExtension,
+  postgresqlTestExtension,
+  postgresqlPackages,
+  python3,
+  readline,
+  stdenv,
+  zlib,
+}:
+
+let
+  # TODO: I'll remove this as well, just testing stuff
+  extensions =
+    buildEnv {
+      name = "postgresql-custom";
+      paths = [ postgresql postgresqlPackages.plpython3 ];
+    };
+in
+postgresqlBuildExtension (finalAttrs: {
+  pname = "omnigres";
+  version = "74ab3167d32e47fec790bb54a979780f7f9f1a28";
+
+  src = fetchFromGitHub {
+    owner = "omnigres";
+    repo = "omnigres";
+    rev = finalAttrs.version;
+    hash = "sha256-PuxfD4XJv6ZxD1LnF3OopY+m057eBT1ZvEfRSbbIyzQ=";
+  };
+
+  nativeBuildInputs = [ cmake flex libtool pkg-config ];
+  buildInputs =
+    postgresql.buildInputs ++ [
+      bison
+      boost
+      brotli
+      bzip2
+      doxygen
+      flex
+      git
+      libcap
+      libuv
+      libxml2
+      lld
+      netcat
+      openssl
+      perl
+      python3
+      readline
+      zlib
+      # TODO: just some hacky tests, remove this later
+      extensions
+    ];
+
+  preConfigure = ''
+    substituteInPlace cmake/PostgreSQLExtension.cmake \
+      --replace "\''${_ext_dir}" "$out"
+  '';
+
+  cmakeFlags = [
+     "-DCMAKE_BUILD_TYPE=Release"
+     "-DPGVER=${postgresql.version}"
+     "-DNETCAT=${netcat}/bin/nc"
+     "-DOPENSSL_CONFIGURED=1"
+     "-DPG_CONFIG=${postgresql.pg_config}/bin/pg_config"
```

Comment:
The intention on Omnigres' side is that only `PG_CONFIG` is necessary for configuring Postges, the rest should be automatically figured out. If it doesn't, I'd love to see where it fails.

Reply:
It seems that's case right now, IDK why I was having errors before, keeping only this flag now.